sunpy/physics/differential_rotation.py

```diff
@@ -254,7 +256,15 @@ def diffrot_map(smap, dt, pad=False):
         A map with the result of applying solar differential rotation to the
         input map.
     """
-    new_time = smap.date + timedelta(seconds=dt.to(u.s).value)
+    if time and dt:
```

Comment:
Should this not be more explicit....
if (time is not None) and (dt is not None):